geode-core/src/main/java/org/apache/geode/internal/cache/ValidateCacheServerProfileProcessor.java

```diff
@@ -0,0 +1,312 @@
+            dm);
+      }
+    }
+
+    private String checkCacheServerProfileCompatibility(LocalRegion localRegion,
+        CacheDistributionAdvisor.CacheProfile cacheProfile) {
+      String cspResult = null;
+      Map<String, CacheServiceProfile> myProfiles = localRegion.getCacheServiceProfiles();
+      // Iterate and compare the remote CacheServiceProfiles to the local ones
+      for (CacheServiceProfile remoteProfile : cacheProfile.cacheServiceProfiles) {
+        CacheServiceProfile localProfile = myProfiles.get(remoteProfile.getId());
+        if (localProfile != null) {
+          cspResult = remoteProfile.checkCompatibility(localRegion.getFullPath(), localProfile);
+        }
+        if (cspResult != null) {
+          return cspResult;
+        }
+      }
+      return cspResult;
+    }
+
+    public int getDSFID() {
+      return VALIDATE_CACHE_PROFILE_MESSAGE;
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder buff = new StringBuilder();
+      buff.append(this.getClass().getName() + " (adviseePath=");
+      buff.append(this.adviseePath);
+      buff.append("; processorId=");
+      buff.append(this.processorId);
+      buff.append("; profile=");
+      buff.append(this.profile);
+      buff.append(")");
+      return buff.toString();
+    }
+
+    @Override
+    public void fromData(DataInput in) throws IOException, ClassNotFoundException {
+      super.fromData(in);
+      this.adviseePath = DataSerializer.readString(in);
+      this.processorId = in.readInt();
+      // set the processor ID to be able to send reply to sender in case of any
+      // unexpected exception during deserialization etc.
+      ReplyProcessor21.setMessageRPId(this.processorId);
+      this.profile = DataSerializer.readObject(in);
+    }
+
+    @Override
+    public void toData(DataOutput out) throws IOException {
+      super.toData(out);
+      DataSerializer.writeString(this.adviseePath, out);
+      out.writeInt(this.processorId);
+      DataSerializer.writeObject(this.profile, out);
+    }
+  }
+
+  public static class ValidateCacheServerProfileReplyMessage extends ReplyMessage {
+
+    public static void send(InternalDistributedMember recipient, int processorId,
+        ReplyException exception, ClusterDistributionManager dm) {
+      ValidateCacheServerProfileReplyMessage m = new ValidateCacheServerProfileReplyMessage();
+
+      m.processorId = processorId;
+      if (exception != null) {
+        m.setException(exception);
+        if (logger.isDebugEnabled()) {
+          logger.debug("Replying with exception: {}" + m, exception);
+        }
+      }
+      m.setRecipient(recipient);
+      dm.putOutgoing(m);
+    }
+
+    @Override
+    public int getDSFID() {
+      return VALIDATE_CACHE_SERVER_REPLY_MESSAGE;
+    }
+
+    @Override
+    public void fromData(DataInput in) throws IOException, ClassNotFoundException {
```

Comment:
probably don't need to override from and to data calls since they arent changing anything from the base